### PR TITLE
Fix broken go upgrade script

### PR DESCRIPTION
**Pull request by inteon**
Refactors the sha-learning and upgrade scripts, fixing the go-bump script that did not yet update the go sha.

### CODE REVIEW

File: scripts/patch_go_version.sh [hunk 1]
```diff
@@ -26,13 +50,15 @@ if [[ $NEW_VERSION == "" ]]; then
...
-if [[ $NEW_VERSION == $LOCAL_VERSION ]]; then
+if [[ "$NEW_VERSION" == "$LOCAL_VERSION" ]]; then
 	echo "go version is up to date"
 	exit 0
 fi
 
-TARGET=modules/tools/00_mod.mk
+echo "updating go version to $NEW_VERSION in $tools_makefile"
+
+sed -i "s/^VENDORED_GO_VERSION := $LOCAL_VERSION$/VENDORED_GO_VERSION := $NEW_VERSION/" "${script_dir}/${tools_makefile}"
```
[the remark this answers is not in the document]
Comment (inteon) [reply scripts/patch_go_version.sh:60]: Thanks for spotting this, will update the script.

File: Makefile [hunk 2]
```diff
@@ -21,46 +21,68 @@
...
+#
+# See https://www.gnu.org/software/make/manual/html_node/Secondary-Expansion.html
+.SECONDEXPANSION:
+
+# For details on some of these "prelude" settings, see:
+# https://clarkgrubb.com/makefile-style-guide
 MAKEFLAGS += --warn-undefined-variables --no-builtin-rules
 SHELL := /usr/bin/env bash
-.SHELLFLAGS := -eu -o pipefail -c
+.SHELLFLAGS := -uo pipefail -c
```
Comment (inteon) [inline Makefile:35]: Changed this to match the config described here: https://github.com/cert-manager/makefile-modules/blob/main/modules/repository-base/base/Makefile

Comment (SgtCoDFish) [inline Makefile:35]: note: this is changed to match https://github.com/cert-manager/makefile-modules/blob/140169ff41d87878ddb0fdfd5ecf567aee25d992/modules/repository-base/base/Makefile which is used in other repos. It's not really required and not really related to the rest of the PR but it's done for conformity.

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve

Tested locally on macOS and it worked!

**Pull request merged**